src/db/transformer.ts

```diff
@@ -13,22 +13,19 @@ import {
   IndexRecord,
 } from './commit';
 import type {Hash} from '../hash';
-import * as dag from '../dag/mod';
+import type * as dag from '../dag/mod';
 import type {ReadonlyJSONValue} from '../json';
 import {HashRefType} from './hash-ref-type';
 import type {Value} from '../kv/store';
 
 type OldHash = Hash;
 type NewHash = Hash;
 
-export class Transformer<Tx = dag.Write> {
-  private readonly _tx: Tx;
-  private readonly _transforming: Map<Hash, Promise<Hash>> = new Map();
+export abstract class BaseTransformer {
+  private readonly _transforming: Map<OldHash, Promise<NewHash>> = new Map();
   private readonly _writtenMappings = new Map<OldHash, NewHash>();
 
-  constructor(tx: Tx) {
-    this._tx = tx;
-  }
+  // no constructor as the base class doesn't take a dag.Read or dag.Write
```

Comment:
i don't think this comment is needed in the checked in code.

Reply:
I thought it was pretty useful :-)